### PR TITLE
Indicate EPOLLRDHUP when the peer closed the connection

### PR DESCRIPTION
**Pull request by jschlumpp**
The RDHUP event is used by nginx to reduce the amount of syscalls it has to do by knowing whether a EOF is coming up. Not setting this event will cause nginx to hang with fcgi_pass configurations.

### CODE REVIEW

File: sockets.c [hunk 1]
```diff
@@ -421,16 +423,25 @@ get_lwip_socket_events(struct lwip_sock *sock)
...
+			return events;
+		}
 
-		return events;
+                /* Check whether the TCP connection is in CLOSE_WAIT (= got a
+                 * FIN packet). In that case the peer will not sent more data
+                 * and we can tell the application that the receive buffer is
+                 * everything we got until the EOF.
+                 */
+                if (sock->conn->pcb.tcp->state == CLOSE_WAIT)
```
Comment (michpappas) [inline sockets.c]: This one needs to additionally check if `sock->conn.pcb.tcp == NULL` as `pcb` is freed by lwip if the connection is reset: https://github.com/unikraft/fork-lwip/blob/UNIKRAFT-2_1_x/src/core/tcp_in.c#L442

**Review by michpappas — CHANGES_REQUESTED** [review]

**Review by michpappas — APPROVED** [review]
Reviewed-by: Michalis Pappas <michalis@unikraft.io>

**Review by eduardvintila — APPROVED** [review]
Thanks.

Reviewed-by: Eduard Vintilă <eduard.vintila47@gmail.com>

**Review by mariapana — APPROVED** [review]
All good!

Reviewed-by: Maria Pană <maria.pana4@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**